### ISSUE TITLE
tuya TS004F does nothing

### ISSUE THREAD

**Issue by QKnoliver** [kkossev/SmartThings#1]
ive tested the groovy but with no luck of getting a valid response for a button pressing.

**Comment by kkossev** [comment]
Hello,

Did the device pair successfully (probably just as a 'thing')?
If yes, then check at the web portal whether the DH is the correct one (My devices -> device name -> Edit ) : 
![image](https://user-images.githubusercontent.com/6189950/120887898-2b139200-c5fe-11eb-9e6f-3aebf08395db.png)

Then press on 'List Events' :
![image](https://user-images.githubusercontent.com/6189950/120887939-5bf3c700-c5fe-11eb-8ef4-9699419fb689.png)

When you press a button, the events should be listed there:

![image](https://user-images.githubusercontent.com/6189950/120888032-d1f82e00-c5fe-11eb-98cc-e2c1ad8f73b9.png)


Do you see any events on this page?

Regards!

**Comment by QKnoliver** [comment]
Sorry my bad. It seems to work. But there is only visual response of button presses after you ad the first action. So thats why I thought it didn't work when I just added it.

What I do here is that there seems to be complication with the hub v2.

I have th v3 so I can't test

**Comment by kkossev** [comment]
It is quite possible that this thing may not work on ST v2. My ST hub is v3 also.

I tried initially on Hubitat Elevation hub, but no luck there - the messages that are decoded from HE radio do not give enough information to distinguish between the actions of the different buttons.
I failed also to pair TS004F to Home Assistant, although others report success there. But I am a very newbie on the HomeAssistant platform and maybe doing something wrong there.

Today I had success decoding the two right buttons long-press (held) and release actions. I will update here my test DH some hours later.

**Comment by QKnoliver** [comment]
Okay but someone has reported that it doesn't work with his hub v3 the button won't connect so it gives no device details with first contact. So the event list is empty. Any idea for this. 

And awesome man hope to here about the update soon

**Comment by kkossev** [comment]
Just pushed  v.1.1 - added 'held' and 'up_hold' (release) events decoding for the 2 right buttons (3 and 4).

@QKnoliver can you test it on your platform?
I use the 'Advanced Button Controller' SmartThings app to assign actions to the scene switch device, including 'held'
![image](https://user-images.githubusercontent.com/6189950/120930018-23ceb000-c6f4-11eb-8f43-9acd196f911d.png)

**Comment by QKnoliver** [comment]
in the app i cant assign actions but in the events it is registered only long press for 3 and  4, and just one press for 1,2,3,4.

i dont seem to see an good interaction with the hub.

![image](https://user-images.githubusercontent.com/81033656/120930347-398ca700-c6ed-11eb-91ff-2797ca5442e9.png)
![image](https://user-images.githubusercontent.com/81033656/120930426-78226180-c6ed-11eb-8ee0-76b5ee1b5abc.png)
![image](https://user-images.githubusercontent.com/81033656/120930462-9ab47a80-c6ed-11eb-8d3f-4c3af4fe2f33.png)

**Comment by kkossev** [comment]
OK, thank you for your cooperation!

As I mentioned above, I was using the ABC Controller application to interact with other devices in the HUB ( I used an OSRAM zigbee bulb to test with ).

But my goals are different, as I have migrated to Hubitat almost a year ago and I am using my SmartThings hub only for few devices, that are not supported by Hubitat. Such devices are the IKEA buttons controllers, and now this weird 'Powered by Tuya' scene switch... So for my goals it is more convenient to have just one device, sending button events with different numbers (1,2,3,4), because this way it is easier to integrate with Hubitat ( and also node-red that I started to use lately for my automation).

So my advice for you would be to stick with the  "SangBoy" /  "YooSangBeom" driver. This is the device handler that handles also other model scene switches and is actively maintained by the author.

Best regards and thank you again for your cooperation and help! Good luck!
